packages/services/src/view-service/fees.ts

```diff
@@ -3,13 +3,19 @@ import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_
 
 // Attempts to extract a fee token from the assets used in the actions of the planner request
 // Priority in descending order
-export const extractAltFee = (request: TransactionPlannerRequest): AssetId => {
+export const extractAltFee = (
+  request: TransactionPlannerRequest,
+  stakingTokenAssetId: AssetId,
+): AssetId => {
   const outputAsset = request.outputs.map(o => o.value?.assetId).find(Boolean);
   if (outputAsset) return outputAsset;
 
   const swapAsset = request.swaps.map(assetIn => assetIn.value?.assetId).find(Boolean);
   if (swapAsset) return swapAsset;
 
+  const ics20Withdrawl = request.ics20Withdrawals.find(Boolean);
+  if (ics20Withdrawl) return stakingTokenAssetId;
```

Comment:
I'm a little confused about the logic here. If a request has any ics20Withdrawal, it should use the staking token instead of an alternative fee? But if they ended up in `extractAltFee()` it guarantees they don't have the native token. So their transaction will error.

Reply:
== Pairing discussion ==

```ts
  const ics20WithdrawAsset = request.ics20Withdrawals.map(w => w.denom).find(Boolean);
  if (ics20WithdrawAsset) return assetIdFromBaseDenom(ics20WithdrawAsset.denom);
```